### ISSUE TITLE
Rate Limiting after 20 refreshes

### ISSUE THREAD

**Issue by darthfishious** [bombardier-gif/covid-vaccine-booking#448]
Getting rate limited after running the script for a little more than 2 minutes every single time. tried pinging only a single district with multiple refresh rates ranging from 5 to 15, with the same result. did they make some changes in the backend?

**Comment by hamza-u** [comment]
+1

**Comment by shivaslifzz** [comment]
rate limit is set 10 refreshs after that it will throw rate limit issue @bombardier-gif @6mxo89sqh @shailesh @BumpyKnuckls

**Comment by BaasilSiddiqui** [comment]
+1

**Comment by darthfishious** [comment]
they have even got rid of captcha. they are making changes

**Comment by WojtekBear93** [comment]
getting ratelimited after 20 calender searches irrespective of the search frequency.
observed the same in 8, 10, 12, 15 seconds

**Comment by shivaslifzz** [comment]
now they changed it from 10 to 20 refreshes

**Comment by manasshukla1** [comment]
+1

**Comment by nolimits0007** [comment]
Same Here.

<img width="952" alt="Screenshot 2021-06-03 at 19 24 51" src="https://user-images.githubusercontent.com/84738436/120656766-867c3d80-c4a1-11eb-8add-1acd52ca43be.png">

**Comment by lohithvadde** [comment]
Tried with my own python script and this is what i got after 20 searches.

status code: 403
response: {"message":"User is not authorized to access this resource with an explicit deny"}

This is a huge blow from COWIN !

May be we all need to enter OTP (but don't click enter) and wait for the telegram alert to kick in.

**Comment by subashjain** [comment]
> Tried with my own python script and this is what i got after 20 searches.
> 
> status code: 403
> response: {"message":"User is not authorized to access this resource with an explicit deny"}
> 
> This is a huge blow from COWIN !
> 
> May be we all need to enter OTP (but don't click enter) and wait for the telegram alert to kick in.

OTP WILL EXPIRE IN 3 MINUTES

**Comment by lohithvadde** [comment]
> > Tried with my own python script and this is what i got after 20 searches.
> > status code: 403
> > response: {"message":"User is not authorized to access this resource with an explicit deny"}
> > This is a huge blow from COWIN !
> > May be we all need to enter OTP (but don't click enter) and wait for the telegram alert to kick in.
> 
> OTP WILL EXPIRE IN 3 MINUTES

Yes. Of course, you need to run the bot as per some pattern. Eg: Kokilaben hospital always opens at 4pm in Mumbai.

**Comment by 6mxo89sqh** [comment]
Is this refresh resuming ever ? After 10 minutes? Next login etc?

**Comment by WojtekBear93** [comment]
> Is this refresh resuming ever ? After 10 minutes? Next login etc?

yes, refer
https://github.com/bombardier-gif/covid-vaccine-booking/issues/448#issuecomment-853878348

**Comment by shivaslifzz** [comment]
this issue is fixed by #449 but it takes time to merge

**Comment by shivaslifzz** [comment]
> this issue is fixed by #449 but it takes time to merge

but it's using the public API

**Comment by darthfishious** [comment]
can we force expire the OTP after 19 refreshes and request new one? i've been trying to test this by using multiple instances of the script and starting each instance after completing 19 refreshes of the previous instance, and facing no problem. each new instance is able to request OTP even if the previous instance was refreshing at 1 second. and then it continues with 19 refreshes without any problem, even if i cross 100 api calls in 5 minutes

**Comment by shivaslifzz** [comment]
> can we force expire the OTP after 19 refreshes and request new one? i've been trying to test this by running sequential instances of the script and starting each instance after completing 19 refreshes of the previous instance, and facing no problem. each new instance is able to request OTP even if the previous instance was refreshing at 1 second. and then it continues with 19 refreshes without any problem, even if i cross 100 api calls in 5 minutes

it's possible but they may block your account for  requesting too many otps,but I know for IOS we are sending otp for every 3 mins

**Comment by darthfishious** [comment]
so in order to make users request OTP only after 3 minutes, we can put in a request that users should try to use refresh rate greater than 9 seconds. but they should have the option to refresh after 1 second at their own risk, which can be useful for high traffic districts when the slot opening timings are known precisely.

**Comment by shivaslifzz** [comment]
> so in order to make users request OTP only after 3 minutes, we can put in a request that users should try to use refresh rate greater than 9 seconds. but they should have the option to refresh after 1 second at their own risk, which can be useful for high traffic districts when the slot opening timings are known precisely.

![image](https://user-images.githubusercontent.com/61616987/120666725-70bf4600-c4aa-11eb-98f9-749145793a67.png)
change 600 to 180 it will expire after 3 mins

**Comment by darthfishious** [comment]
> > so in order to make users request OTP only after 3 minutes, we can put in a request that users should try to use refresh rate greater than 9 seconds. but they should have the option to refresh after 1 second at their own risk, which can be useful for high traffic districts when the slot opening timings are known precisely.
> 
> ![image](https://user-images.githubusercontent.com/61616987/120666725-70bf4600-c4aa-11eb-98f9-749145793a67.png)
> change 600 to 180 it will expire after 3 mins

set it to 190 and refresh rate of 10 seconds. running without any problem. still think that we should give the option of expiring OTP after 19 refreshes, rather than 3 minutes.

**Comment by shivaslifzz** [comment]
> now they changed it from 10 to 20 refreshes

again they changed the refreshs from 20 to 10

**Comment by Nakul93** [comment]
> > so in order to make users request OTP only after 3 minutes, we can put in a request that users should try to use refresh rate greater than 9 seconds. but they should have the option to refresh after 1 second at their own risk, which can be useful for high traffic districts when the slot opening timings are known precisely.
> 
> ![image](https://user-images.githubusercontent.com/61616987/120666725-70bf4600-c4aa-11eb-98f9-749145793a67.png)
> change 600 to 180 it will expire after 3 mins

I think we should find another way, since expiring OTP after 3mins is a hassle for ios users especially since they have to manually run the automation

**Comment by Saurabh-meshram** [comment]
This happened because of cowin booking chrome extension. A guy developed chrome extension and automated captcha as well. All leading news portals reported this and govt took action by removing captcha and also reduced refresh rate. :(

**Comment by WojtekBear93** [comment]
> > now they changed it from 10 to 20 refreshes
> 
> again they changed the refreshes from 20 to 10

no it is still 20

**Comment by shivaslifzz** [comment]
> > > now they changed it from 10 to 20 refreshes
> > 
> > 
> > again they changed the refreshes from 20 to 10
> 
> no it is still 20

I guess they are playing with the values

**Comment by Saurabh-meshram** [comment]
> > > > now they changed it from 10 to 20 refreshes
> > > 
> > > 
> > > again they changed the refreshes from 20 to 10
> > 
> > 
> > no it is still 20
> 
> I guess they are playing with the values

Any idea how telegram notification still working despite rate limit ?

**Comment by shivaslifzz** [comment]
> > > > > now they changed it from 10 to 20 refreshes
> > > > 
> > > > 
> > > > again they changed the refreshes from 20 to 10
> > > 
> > > 
> > > no it is still 20
> > 
> > 
> > I guess they are playing with the values
> 
> Any idea how telegram notification still working despite rate limit ?

they might be using public api's or rerouting the IP when ever they reach rate limit

**Comment by thewebcapsule** [comment]
Will the booking work since the captcha is removed now?

**Comment by shivaslifzz** [comment]
> Will the booking work since the captcha is removed now?

today 9:10 only we made a booking

**Comment by thewebcapsule** [comment]
> > Will the booking work since the captcha is removed now?
> 
> today 9:10 only we made a booking

Did you select automate captcha as yes?

**Comment by shivaslifzz** [comment]
> > > Will the booking work since the captcha is removed now?
> > 
> > 
> > today 9:10 only we made a booking
> 
> Did you select automate captcha as yes?

yes

**Comment by Saurabh-meshram** [comment]
> Will the booking work since the captcha is removed now?

comment out captcha related code.
https://github.com/pallupz/covid-vaccine-booking/commit/1712fc62521620eb0dc28008da23bf1e2a095b14

**Comment by nolimits0007** [comment]
With latest restrictions-on-refresh #449  I selected Covaxin but it booked Covishield. Plz Help.

<img width="828" alt="Screenshot 2021-06-03 at 21 51 51" src="https://user-images.githubusercontent.com/84738436/120685377-635f8700-c4bd-11eb-87a9-037e22feb36a.png">

**Comment by warlock2k** [comment]
This happened to me as well, need to create a separate issue.

**Comment by shailesh** [comment]
> Will the booking work since the captcha is removed now?

It will work because None value or any value returned by Captcha won't be used in next booking API so yes it will work. But better to comment it for time saving

**Comment by 40ZqTYY4ag3tAO4hdlxR1** [comment]
> > so in order to make users request OTP only after 3 minutes, we can put in a request that users should try to use refresh rate greater than 9 seconds. but they should have the option to refresh after 1 second at their own risk, which can be useful for high traffic districts when the slot opening timings are known precisely.
> 
> ![image](https://user-images.githubusercontent.com/61616987/120666725-70bf4600-c4aa-11eb-98f9-749145793a67.png)
> change 600 to 180 it will expire after 3 mins

SO what happens if we find an appointment on the 18th call, do we get blocked when making the booking, which takes 1?/2?/3?more calls to BOOKING_URL ?

**Comment by arpit0808** [comment]
What is the full working version ? I tried with PR#452 and on AWS was stuck in  AWS Re-assignment loop. I had used Private IP

**Comment by 6mxo89sqh** [comment]
@arpit0808 'search for seven days' will be stuck in aws re-assignment loop. please use fixed date search.

**Comment by arpit0808** [comment]
@6mxo89sqh :Thanks a lot .Works like a charm now

**Comment by nolimits0007** [comment]
Can u we don't have all date option by any means.
